sktime/classification/shapelet_based/_stc.py

```diff
@@ -84,11 +79,26 @@ def fit(self, X, y):
         -------
         self : object
          """
-        X, y = check_X_y(X, y, enforce_univariate=True)
-        self.n_classes = np.unique(y).shape[0]
-        self.classes_ = class_distribution(np.asarray(y).reshape(-1, 1))[0][0]
+        _X, _y = check_X_y(X, y, enforce_univariate=True)
 
-        self.classifier.fit(X, y)
+        # if y is a pd.series then convert to array.
+        if isinstance(_y, pd.Series):
+            _y = _y.to_numpy()
+
+# generate pipeline in fit so that random state can be propogated properly.
+        self.classifier = Pipeline([
```

Comment:
I'd call it `self.classifier_`, by sklearn convention, attributes changed in fit have a trailing underscore